contracts/WGHST/WGHST.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+/// @notice Minimalist and modern Wrapped Ether implementation.
+/// @author Solmate (https://github.com/transmissions11/solmate/blob/main/src/tokens/WETH.sol)
+/// @author Inspired by WETH9 (https://github.com/dapphub/ds-weth/blob/master/src/weth9.sol)
+contract WGHST is ERC20("Wrapped GHST", "WGHST", 18) {
+    using SafeTransferLib for address;
+
+    event Deposit(address indexed from, uint256 amount);
+
+    event Withdrawal(address indexed to, uint256 amount);
+
+    receive() external payable virtual {
+        deposit();
+    }
+
+    function deposit() public payable virtual {
+        _mint(msg.sender, msg.value);
+
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    function withdraw(uint256 amount) public virtual {
```

Comment:
@cinnabarhorse please update wghst as you test in socket bridging

Reply:
@orionstardust Note that the WGHST contract is different from the UnwrapSuperToken. We're not using the same contract

Reply:
Oh, I thought wrong.